src/lib/marshal/workflow/writer.ts

```diff
@@ -312,15 +305,72 @@ const buildWorkflowDirBundle = (
       // bundle for writing to the file system later. Then replace the field
       // content with the extracted file path and mark the field as extracted
       // with @ suffix.
+      //
       // TODO: Consider guarding against an edge case, and check if the relpath
       // already exists in the bundle, and if so make the relpath unique.
       set(bundle, [relpath], content);
       set(step, `${objPathStr}${FILEPATH_MARKER}`, relpath);
       unset(step, objPathParts);
     }
+
+    // Lastly, recurse thru any branches that exist in the workflow tree
+    if (step.type === StepType.IfElse) {
+      for (const branch of step.branches) {
+        recursivelyBuildWorkflowDirBundle(
+          bundle,
+          branch.steps,
+          localWorkflowStepsByRef,
+        );
+      }
+    }
   }
+};
+
+/*
+ * For a given workflow payload (and its local workflow reference), this function
+ * builds a "workflow directory bundle", which is an obj made up of all the
+ * relative file paths (within the workflow directory) and its file content to
+ * write the workflow directory.
+ *
+ * Every workflow will always have a workflow.json file, so every bundle includes
+ * it and its content at minimum. To the extent the workflow includes any
+ * extractable fields, those fields content get extracted out and added to the
+ * bundle.
+ *
+ * Important things to keep in mind re: content extraction:
+ * 1. There can be multiple places in workflow json where content extraction
+ *    happens.
+ * 2. There can be multiple levels of content extraction happening, currently
+ *    at a maximum of 2 levels.
+ *
+ * The way this function works and handles the content extraction is by:
+ * 1. Traversing the given step node, and compiling all annotated extraction
+ *    settings by the object path in the node *ordered from leaf to root*.
+ * 2. Iterate over compiled extraction settings from leaf to root, and start
+ *    extracting out the field as needed. In case the node that needs to be
+ *    extracted out contains extracted file paths, then those file paths get
+ *    rebased to relative to the referenced file.
+ */
+const buildWorkflowDirBundle = (
+  remoteWorkflow: WorkflowData<WithAnnotation>,
+  localWorkflow: AnyObj = {},
+): WorkflowDirBundle => {
+  const bundle: WorkflowDirBundle = {};
+  const mutWorkflow = cloneDeep(remoteWorkflow);
+
+  const localWorkflowStepsByRef = keyLocalWorkflowStepsByRef(
+    (localWorkflow.steps as AnyObj[]) || [],
```

Comment:
```suggestion
    localWorkflow.steps
```
Since we are now typing this as `unknown` above.